### ISSUE TITLE
CertificateIdentity force-unwrap crashes QuickSign on iOS 17

### ISSUE THREAD

**Issue by AlexTech01** [QuickSign-Team/QuickSign#2]
On iPadOS 17.4RC, QuickSign crashes upon trying to sign an app due to line 26 of Hoster.swift (the line involving `CertificateIdentity`). For whatever reason, on iPadOS 17.4RC (and potentially other versions), the `identity` is null, and force-unwrapping it causes a crash. I haven't experienced this on my other devices running iOS 18.1 beta 1, iPadOS 18.0 beta 1, and visionOS 2.0 beta 5 (in iPad compatibility mode).

Things I've tried that didn't work:
-Making a self-signed cert for `localhost.direct`, giving it a password or not, and trusting it in the device settings
-Modifying the existing cert for `localhost.direct` to have a different password or no password
-Uninstalling QuickSign and erasing the certificate passwords before reinstalling
-Restarting the device

Things I've tried that did work:
-Getting rid of the `caCertificate` and `identity` vars (and their respective url vars) and invoking `Server()` with no arguments. This, however, causes it to not have HTTPS, so the on-device manifest option won't work. This does fix the server manifest on the device, though. If no true solution can be found, checking if `identity` is null and then forcing server manifests only without a local HTTPS server might work as a solution.

**Comment by c22dev** [comment]
Hi, thanks for opening the issue.
Please note that you are probably building the app off commits that aren't fully functional, and we can't offer proper support for now.

This is a weird issue; none of the team members are experiencing this, and I can't seem to be able to reproduce it. 
Anything else in the console? I assume the app was installed using Xcode, right ? If not, please tell me.

Would you mind adding me on Discord so we can see more in detail ? My id is `c22dev`.

**Comment by AlexTech01** [comment]
Yes, the app was installed through Xcode. Upon further testing, I can reproduce this issue with a brand new iOS 17.4 and 17.5 Xcode simulator, so it probably isn't just my device. The 18.0 simulator and all my 18.0/18.1 devices don't have the same issue for some reason.

```text
/var/mobile/Containers/Data/Application/<app-uuid>/Documents/Apple Distribution: <my team>.p12
>>> Signing: 	/var/mobile/Containers/Data/Application/<app-uuid>/Documents/temp/tempext/Payload/Delta.app ...
>>> AppName: 	Delta
>>> BundleId: 	com.rileytestut.Delta
>>> BundleVer: 	102
>>> TeamId: 	<my teamid>
>>> SubjectCN: 	Apple Distribution: <my team> (<team id>)
>>> ReadCache: 	NO
>>> SignFolder: Frameworks/Systems.framework, (Systems)
[33m>>> No Enough CodeSignature Space. Length => Now: 51872, Need: 62924
[0m[33m>>> Realloc CodeSignature Space... 
[0m[33m>>> Success!
[0m>>> SignFile: 	Frameworks/libswift_Concurrency.dylib
>>> SignFolder: Delta.app, (Delta)
[33m>>> No Enough CodeSignature Space. Length => Now: 167680, Need: 251013
[0m[33m>>> Realloc CodeSignature Space... 
[0m[33m>>> Success!
[0mSign successful
plist data incomming !!!!! 
 731 bytes
QuickSign/Hoster.swift:26: Fatal error: Unexpectedly found nil while unwrapping an Optional value
```

This example is a log from trying to install Delta, but the same error occurs with every app.

**Comment by c22dev** [comment]
This is really weird.
Ig you are using latest commit ?
I'll get an iOS 17 device around and will mess with it...

**Comment by AlexTech01** [comment]
Yes, using latest commit.

**Comment by c22dev** [comment]
Wow, this is definitely really weird haha
I'm downloading an iOS 17 simulator and I'll mess around with it. Will keep you up if I find the solution.

**Comment by c22dev** [comment]
This seems to be an issue with Telegraph's certificate handling, and not linked to QuickSign.
I can manually import the p12 from app's bundle into the macOS keychain without any issue.

Apparently, [the exact same issue happened to someone else](https://github.com/Building42/Telegraph/issues/143) but they didn't provide any solutions.

Again, could you add me on Discord ? My username is `c22dev`.

**Comment by alwwwd** [comment]
Same on iOS 17.3.1

**Comment by c22dev** [comment]
Fixed in [be58dc3](https://github.com/QuickSign-Team/QuickSign/commit/be58dc38df864e5629052273a0143c26469f8828).

Please make sure to Clean Build Folder, and to use (for now) the 2 supported servers:
- c22dev's (CH)
- Render Dummy Server (DE)

You can change this in Settings>App Installation>Server.

Tested on iOS 16.2.
